pkg/promlib/converter/prom.go

```diff
@@ -39,7 +39,7 @@ func ReadPrometheusStyleResult(jIter *jsoniter.Iterator, opt Options) backend.Da
 l1Fields:
 	for l1Field, err := iter.ReadObject(); ; l1Field, err = iter.ReadObject() {
 		if err != nil {
-			return rspErr(fmt.Errorf("response from prometheus couldn't parsed. it is non-json"))
+			return rspErr(fmt.Errorf("response from prometheus couldn't parsed. it is non-json: %w", err))
```

Comment:
This will not show us the actual error. It will be the error from parsing the json. We are still missing the information that is in res.body. That is assumed to be a string with some information, and as you mentioned, could be big. But, this is what was requested, to return the information from the response, not the error from parsing json. 🤷🏻 

Do we have a local reproduction of this error showing examples of the non-json response? If it is a bunch of html, could we strip the html and just return the text in the actual response?